src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/Egress/S3/MultiPartUploadStreamTests.cs

```diff
@@ -0,0 +1,117 @@
+            {
+                await stream.WriteAsync(bytes, 0, bytes.Length);
+                allBytes.AddRange(bytes);
+            }
+
+            await stream.FinalizeAsync(CancellationToken.None);
+
+            Assert.Equal(3, stream.Parts.Count);
+            Assert.True(_s3.Uploads("bucket").TryGetValue(uploadId, out var data));
+            Assert.Equal(3, data.Count - 1); // the first entry contains meta information
+
+            Assert.Equal(MultiPartUploadStream.MinimumSize, data[1].Size);
+            Assert.Equal(allBytes.Take(MultiPartUploadStream.MinimumSize).ToArray(), data[1].Bytes());
+
+            Assert.Equal(MultiPartUploadStream.MinimumSize, data[2].Size);
+            Assert.Equal(allBytes.Skip(MultiPartUploadStream.MinimumSize).Take(MultiPartUploadStream.MinimumSize).ToArray(), data[2].Bytes());
+
+            Assert.Equal(allBytes.Skip(MultiPartUploadStream.MinimumSize * 2).ToArray(), data[3].Bytes());
+            Assert.Equal(MultiPartUploadStream.MinimumSize - 1024, data[3].Size);
+        }
+
+        [Fact]
+        public async Task ItShouldWriteSinglePartialPart()
+        {
+            var uploadId = (await _s3.InitiateMultipartUploadAsync("bucket", "key")).UploadId;
+            await using var stream = new MultiPartUploadStream(_s3, "bucket", "key", uploadId, 1024);
+
+            var allBytes = new List<byte>();
+            foreach (var bytes in WithBytesReturned(MultiPartUploadStream.MinimumSize - 1024))
+            {
+                await stream.WriteAsync(bytes, 0, bytes.Length);
+                allBytes.AddRange(bytes);
+            }
+
+            await stream.FinalizeAsync(CancellationToken.None);
+
+            Assert.Single(stream.Parts);
+            Assert.True(_s3.Uploads("bucket").TryGetValue(uploadId, out var data));
+            Assert.Equal(1, data.Count - 1); // the first entry contains meta information
+            Assert.Equal(MultiPartUploadStream.MinimumSize - 1024, data.Last().Size);
+            Assert.Equal(allBytes.ToArray(), data.Last().Bytes());
+        }
+
+        [Fact]
+        public async Task ItShouldWriteSingleFulllPart()
+        {
+            var uploadId = (await _s3.InitiateMultipartUploadAsync("bucket", "key")).UploadId;
+            await using var stream = new MultiPartUploadStream(_s3, "bucket", "key", uploadId, 1024);
+
+            var allBytes = new List<byte>();
+            foreach (var bytes in WithBytesReturned(MultiPartUploadStream.MinimumSize))
+            {
+                await stream.WriteAsync(bytes, 0, bytes.Length);
+                allBytes.AddRange(bytes);
+            }
+
+            await stream.FinalizeAsync(CancellationToken.None);
+
+            Assert.Single(stream.Parts);
+            Assert.True(_s3.Uploads("bucket").TryGetValue(uploadId, out var data));
+            Assert.Equal(1, data.Count - 1); // the first entry contains meta information
```

Comment:
```suggestion
            StorageData storageItem = Assert.Single(data.Skip(1)); // the first entry contains meta information
```

Allows you to validate the that a single datum is expected and then you get a reference to that item without having to do `data.Last()` all of the time.